From 378fee6141236d43622499b31208ad3d0aff05ea Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Wed, 26 Jun 2013 14:41:47 +0900
Subject: [PATCH 14/21] dfbvideosink: invoke SetConfiguration regardless of
 video mode setting

Whether SetConfiguration() method should be invoked regardless of
the result of gst_dfbvideosink_get_best_vmode(), since the two are
unrelated.
---
 ext/directfb/dfbvideosink.c | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index fd24302..e1eb1dc 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1174,6 +1174,7 @@ gst_dfbvideosink_setcaps (GstBaseSink * bsink, GstCaps * caps)
   if (dfbvideosink->dfb) {
     DFBResult ret;
     GstDfbVMode vmode;
+    DFBDisplayLayerConfig lc;
 
     GST_DEBUG_OBJECT (dfbvideosink, "trying to adapt the video mode to video "
         "geometry");
@@ -1182,7 +1183,6 @@ gst_dfbvideosink_setcaps (GstBaseSink * bsink, GstCaps * caps)
     if (gst_dfbvideosink_get_best_vmode (dfbvideosink,
             GST_VIDEO_SINK_WIDTH (dfbvideosink),
             GST_VIDEO_SINK_HEIGHT (dfbvideosink), &vmode)) {
-      DFBDisplayLayerConfig lc;
       gint width, height, bpp;
 
       width = vmode.width;
@@ -1198,23 +1198,23 @@ gst_dfbvideosink_setcaps (GstBaseSink * bsink, GstCaps * caps)
         GST_WARNING_OBJECT (dfbvideosink, "failed setting video mode %dx%d "
             "at %d bpp", width, height, bpp);
       }
+    }
 
-      lc.flags = DLCONF_PIXELFORMAT;
-      lc.pixelformat = pixel_format;
+    lc.flags = DLCONF_PIXELFORMAT;
+    lc.pixelformat = pixel_format;
 
-      ret = dfbvideosink->layer->SetConfiguration (dfbvideosink->layer, &lc);
-      if (ret != DFB_OK) {
-        GST_WARNING_OBJECT (dfbvideosink, "failed setting layer pixelformat "
-            "to %s", gst_dfbvideosink_get_format_name (pixel_format));
-      } else {
-        dfbvideosink->layer->GetConfiguration (dfbvideosink->layer, &lc);
-        dfbvideosink->out_width = lc.width;
-        dfbvideosink->out_height = lc.height;
-        dfbvideosink->pixel_format = lc.pixelformat;
-        GST_DEBUG_OBJECT (dfbvideosink, "layer %d now configured to %dx%d %s",
-            dfbvideosink->layer_id, lc.width, lc.height,
-            gst_dfbvideosink_get_format_name (lc.pixelformat));
-      }
+    ret = dfbvideosink->layer->SetConfiguration (dfbvideosink->layer, &lc);
+    if (ret != DFB_OK) {
+      GST_WARNING_OBJECT (dfbvideosink, "failed setting layer pixelformat "
+          "to %s", gst_dfbvideosink_get_format_name (pixel_format));
+    } else {
+      dfbvideosink->layer->GetConfiguration (dfbvideosink->layer, &lc);
+      dfbvideosink->out_width = lc.width;
+      dfbvideosink->out_height = lc.height;
+      dfbvideosink->pixel_format = lc.pixelformat;
+      GST_DEBUG_OBJECT (dfbvideosink, "layer %d now configured to %dx%d %s",
+          dfbvideosink->layer_id, lc.width, lc.height,
+          gst_dfbvideosink_get_format_name (lc.pixelformat));
     }
   }
 
-- 
1.8.1.2

